### ISSUE TITLE
[discuss,CI] Checking deterministic codegen across different interpreter runs

### ISSUE THREAD

**Issue by kaushikcfd** [inducer/pytato#286]
Is there a way we can check for deterministic codegen? Challenge(s):
- Within an interpreter run the `PYTHONHASHSEED` is fixed, so the traversals of unordered containers is deterministic. However,  across interpreter runs the `PYTHONHASHSEED` will vary and therefore the unordered containers will have different traversal orders. This may cause non-deterministic loopy codegen.

**Comment by inducer** [comment]
We can just call interpreters in subprocesses with different hash seeds. They could return a pickled kernel via stdout. (base-64 encoded if need be) The MPI tests already do something vaguely like this.

**Comment by kaushikcfd** [comment]
Got a easier solution using multiprocessing:
```python
import multiprocessing as mp


def f():
    a = list(set("abcdefghijkl"))
    print(a)


if __name__ == "__main__":
    ctx = mp.get_context("spawn")
    procs = [ctx.Process(target=f, args=()) for i in range(10)]
    for proc in procs:
        proc.start()

    for proc in procs:
        proc.join()
```

**Comment by inducer** [comment]
Can you set the hash seed though?

**Comment by kaushikcfd** [comment]
Yep, slightly inconvenient that can't set hash seed, but the spawned processes use different hash seeds.

**Comment by matthiasdiener** [comment]
https://stackoverflow.com/a/64138050/1250282 might have an interesting alternative:

```python
import os
os.environ['PYTHONHASHSEED'] = '1234'
os.execv(sys.executable, [sys.executable] + sys.argv)
```

**Comment by kaushikcfd** [comment]
Closed by #311. :)
